golioth/cli.py

```diff
@@ -31,22 +31,38 @@ def __init__(self):
 pass_config = click.make_pass_decorator(Config, ensure=True)
 
 @click.group()
-@click.option('-c', '--config-path', type=Path,
-              help='Path to goliothctl configuration',
-              default=Path.home() / '.golioth' / '.goliothctl.yaml')
-@click.option('--api-key', help='Api key')
+@click.option('-c', '--config-path', type=Path, show_default=True,
+              default=Path.home() / '.golioth' / '.goliothctl.yaml',
+              help='Path to goliothctl configuration.')
+@click.option('--api-key', help='API key to use instead of stored config (--api-project must also be supplied)')
+@click.option('--api-project', default=None, help='Project Name associated with supplied API key')
```

Comment:
API keys are tied to projects, so we can infer the project from the key. Listing projects using the API Key will only provide one project:
```
client = Client(api_key = api_key)
config.default_project = (await client.get_projects())[0]
```

Alternatively, if we don't want to make a call in this function, we could set `config.default_project = None` and then grab the first project in the list from the client in each command.

Reply:
Ah, that makes sense. I didn't understand why the zero index projets being used.

I've updated with your suggestions now, thanks!